core/common/src/main/kotlin/app/k9mail/core/common/mail/EmailDomain.kt

```diff
@@ -0,0 +1,51 @@
+package app.k9mail.core.common.mail
+
+import app.k9mail.core.common.net.Domain
+
+/**
+ * The domain part of an email address.
+ *
+ * @param value String representation of the email domain with the original capitalization.
+ */
+class EmailDomain internal constructor(val value: String) {
+    /**
+     * The normalized (converted to lower case) string representation of this email domain.
+     */
+    val normalized: String = value.lowercase()
```

Comment:
This won't work correctly in the Turkish locale. You need to specify Locale.ROOT.

Reply:
This is not Java. Kotlin's `lowercase()` will do the right thing.
See https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/lowercase.html